js/customize-snapshots.js

```diff
@@ -290,14 +474,15 @@
 			options
 		);
 
+		data = _.extend( args, {
+			nonce: api.settings.nonce.snapshot,
+			customize_snapshot_uuid: component.data.uuid
+		} );
+
 		data = _.extend(
 			{},
 			api.previewer.query(),
-			{
-				nonce: api.settings.nonce.snapshot,
-				customize_snapshot_uuid: component.data.uuid,
-				status: args.status
```

Comment:
We can optimize this extending (3 times to 1). Here, I am merging args with options above, option param contains status and publish date (Optional: in the case of schedule snapshot) so because we don't need to explicitly check for publish date's existence, I am merging it directly.